### PR TITLE
Exosys daemon

### PR DESCRIPTION
**Pull request by Slesarew**
make exosys track events and print them.

soon (days) parser crate upstream would start outputting typed objects

### CODE REVIEW

File: exosys/src/main.rs [hunk 1]
```diff
@@ -89,38 +117,192 @@ async fn dump_finalized_headers(rpc_client: &Client<Config>) -> () {
...
+                    Some(tail) => format!("{}:443{}", &caps["body"], tail.as_str()),
+                    None => format!("{}:443", &caps["body"]),
+                }
+            }
+        }
+        None => str_address.to_string(),
+    }
+}
+
+#[tokio::main]
```
Comment (wigy-opensource-developer) [inline exosys/src/main.rs:148]: Quote from https://docs.rs/tokio/latest/tokio/attr.main.html:

> Note: This macro can be used on any function and not just the main function. Using it on a non-main function makes the function behave as if it was synchronous by starting a new runtime each time it is called. If the function is called often, it is preferable to create the runtime using the runtime builder so the runtime can be reused across calls.

Would not it be simpler to put it once on `async fn main` and `.await` calling these functions from there?

Comment (Slesarew) [reply exosys/src/main.rs:148]: Sure. I've just quickly slapped together chunks of other projects to get the result; I mean to run 2 or 3 threads there concurrently later. It's ugly everywhere at the moment.

**Comment by Slesarew** [comment]
Let's merge it as discussed yesterday?

**Comment by wigy-opensource-developer** [comment]
@Slesarew How comfortable would you be to just create your own forked crates from parity-signer? Vera seemed to do a great job on these SCALE parsers, maybe you could publish it for the whole ecosystem to gain from them.

**Review by wigy-opensource-developer — APPROVED** [review]

**Comment by Slesarew** [comment]
> @Slesarew How comfortable would you be to just create your own forked crates from parity-signer? Vera seemed to do a great job on these SCALE parsers, maybe you could publish it for the whole ecosystem to gain from them. Also, can we get rid of the desub dependency from here?

I'm doing just that, rebranding and forkng (see fork and Alzymologist org). It will take some time though since i need to do everything legally and nicely. Any ideas for new brand name? We can push it to f-droid then finally.

**Pull request merged**